bin/toV2.ts

```diff
@@ -0,0 +1,46 @@
+type Transformer = (options: TransformerParameters) => { key: string, value: unknown };
+type TransformerParameters = {key: string, value: unknown, manifest: any};
+
+const toV2 = (v3) =>
+	Object.entries(v3)
+		.reduce((v2, [key, value]): Record<string, unknown> => {
+			const transformer = transformers[key] ?? identity;
+			const {key: newKey, value: newValue} = transformer({value, key, manifest: v3});
+			return {...v2, [newKey]: newValue};
+		}, {});
+
+const retainKey = (callback: ({value, manifest}) => unknown): Transformer =>
+	({key, value, manifest}) =>
+		({key, value: callback({value, manifest})});
+
+const fixedValue = (value: unknown) => retainKey(() => value);
+
+const identity = retainKey(({value}) => value);
+const transformers: { [key: string]: Transformer } = {
+	manifest_version: fixedValue(2),
+	options_page: fixedValue(undefined),
+	action: ({value}) => ({key: "browser_action", value}),
+	content_security_policy: fixedValue(undefined),
+	key: () => ({
+		key: "browser_specific_settings",
+		value: {
+			gecko: {
+				id: "{ae1ccbaf-b9e4-44e4-b0b1-005db1c0dbd2}"
```

Comment:
what is this lol

Reply:
Remember how we had the `"key"` in our manifest to keep it so we always had the same extension ID? this is the same thing.